docs/mockwbemserver.rst

```diff
@@ -836,389 +926,310 @@ mock environment.
+  parameter exists in that namespace of the mock repository.
+* The CIM properties of a CIM instance specified as an input parameter
+  are defined in the CIM class of the instance and have the correct CIM types.
+* The CIM instance does not yet exist for CreateInstance and does exist for
+  ModifyInstance.
 
-   .. autoautosummary:: pywbem_mock.BaseRepository
-      :methods:
-      :nosignatures:
+The Python operation methods should raise any exceptions using
+:class:`pywbem.CIMError` using the CIM status codes defined in :term:`DSP0200`.
 
-   .. rubric:: Attributes
 
-   .. autoautosummary:: pywbem_mock.BaseRepository
-      :attributes:
+.. _`User-defined instance write providers`:
 
-   .. rubric:: Details
+User-defined instance write providers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+.. automodule:: pywbem_mock._instancewriteprovider
 
-.. autoclass:: pywbem_mock.BaseObjectStore
+.. autoclass:: pywbem_mock.InstanceWriteProvider
    :members:
 
    .. rubric:: Methods
 
-   .. autoautosummary:: pywbem_mock.BaseObjectStore
+   .. autoautosummary:: pywbem_mock.InstanceWriteProvider
       :methods:
       :nosignatures:
 
    .. rubric:: Attributes
 
-   .. autoautosummary:: pywbem_mock.BaseObjectStore
+   .. autoautosummary:: pywbem_mock.InstanceWriteProvider
       :attributes:
 
    .. rubric:: Details
 
 
-.. automodule:: pywbem_mock._inmemoryrepository
-
-.. autoclass:: pywbem_mock.InMemoryRepository
-   :members:
-
-   .. rubric:: Methods
-
-   .. autoautosummary:: pywbem_mock.InMemoryRepository
-      :methods:
-      :nosignatures:
-
-   .. rubric:: Attributes
-
-   .. autoautosummary:: pywbem_mock.InMemoryRepository
-      :attributes:
+.. _`User-defined method providers`:
 
-   .. rubric:: Details
+User-defined method providers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+.. automodule:: pywbem_mock._methodprovider
 
-.. autoclass:: pywbem_mock.InMemoryObjectStore
+.. autoclass:: pywbem_mock.MethodProvider
    :members:
 
    .. rubric:: Methods
 
-   .. autoautosummary:: pywbem_mock.InMemoryObjectStore
+   .. autoautosummary:: pywbem_mock.MethodProvider
       :methods:
       :nosignatures:
 
    .. rubric:: Attributes
 
-   .. autoautosummary:: pywbem_mock.InMemoryObjectStore
+   .. autoautosummary:: pywbem_mock.MethodProvider
       :attributes:
 
    .. rubric:: Details
 
 
-.. index:: pair: User defined providers; user providers
+.. _`Configuration of mocked behavior`:
 
-.. _`User provider interfaces`:
+Configuration of mocked behavior
+--------------------------------
 
-User provider interfaces
-------------------------
+.. automodule:: pywbem_mock.config
+   :members:
 
-.. _`Instance Write Provider`:
 
-Instance Write Provider
-^^^^^^^^^^^^^^^^^^^^^^^
+.. _`Mocker base classes`:
 
-.. automodule:: pywbem_mock._instancewriteprovider
+Mocker base classes
+-------------------
 
-.. autoclass:: pywbem_mock.InstanceWriteProvider
+Some bases classes are included in this documentation in order to provide the
```

Comment:
Andy.   I am not sure but I think that the documentation for these classes pulls in the base class documentation so we should not be including base classes specifically.

Reply:
No. The derived classes have these lists of methods and attributes right after the class itself, and that includes any inherited elements. The element descriptions themselves are not pulled in, so if the base classes are not shown, the links in the attribute and method tables will not be generated as links, but just as bold text. Try it out yourself.

Reply:
Karl agrees.